src/test/java/net/minecraftforge/debug/fluid/CoreFluidTest.java

```diff
@@ -0,0 +1,355 @@
+            {
+                return blockState.getBlock() instanceof ConcretePowderBlock || blockState.getBlock() instanceof FarmBlock;
+            }
+
+            @Override
+            public boolean canBoat(FluidState state, Boat boat)
+            {
+                return false;
+            }
+
+            @Override
+            public void sink(FluidState state, LivingEntity entity) {
+                entity.setDeltaMovement(entity.getDeltaMovement().add(0.0D, (double)-0.1F * entity.getAttribute(net.minecraftforge.common.ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
+            }
+
+            // If the custom fluid is trying to place a flowing block, and it touches a Lava block.
+            // Then it will generate a copy of the BlockState below said position.
+            @Override
+            public boolean handleFluidInteraction(FluidState state, Level level, BlockPos pos)
+            {
+                for (Direction direction : Direction.values())
+                {
+                    if (direction != Direction.DOWN)
+                    {
+                        if (level.getFluidState(pos.relative(direction)).getType() == Fluids.LAVA)
+                        {
+                            level.setBlock(pos, level.getBlockState(pos.below()), Constants.BlockFlags.DEFAULT);
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+        }
+    }
+
+    private static class FluidloggableBlock extends Block implements SimpleWaterloggedBlock
+    {
+        public static final BooleanProperty FLUIDLOGGED = BooleanProperty.create("fluidlogged");
+
+        public FluidloggableBlock(Properties properties)
+        {
+            super(properties);
+            registerDefaultState(getStateDefinition().any().setValue(FLUIDLOGGED, false));
+        }
+
+        @Override
+        protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder)
+        {
+            builder.add(FLUIDLOGGED);
+        }
+
+        @Override
+        public boolean canPlaceLiquid(BlockGetter world, BlockPos pos, BlockState state, Fluid fluid)
+        {
+            return !state.getValue(FLUIDLOGGED) && (state.getFluidState().isEmpty() || state.getFluidState().getType() == fluid);
+        }
+
+        @Override
+        public boolean placeLiquid(LevelAccessor world, BlockPos pos, BlockState blockState, FluidState fluidState) {
+            if (canPlaceLiquid(world, pos, blockState, fluidState.getType()))
+            {
+                if (!world.isClientSide())
+                {
+                    world.setBlock(pos, blockState.setValue(FLUIDLOGGED, true), 3);
+                    world.getLiquidTicks().scheduleTick(pos, fluidState.getType(), fluidState.getType().getTickDelay(world));
+                }
+                return true;
+            }
+            else
```

Comment:
Might as well remove the else from here and just return false so it is more readable as it only takes one line instead of four.